docs/specs/snp-genotyping-vector.md

```diff
@@ -0,0 +1,207 @@
+   variants VCF, regardless of what is in the FILTER column. Therefore
+   the known variants VCF must be created setting FILTER to PASS for
+   all variants to get this to work as desired.
+
+
+### Step: VCF to Zarr conversion
+
+Convert the VCF to Zarr format via the scikit-allel
+[vcf_to_zarr()](https://scikit-allel.readthedocs.io/en/stable/io.html#allel.vcf_to_zarr)
+function. Note the following implementation details:
+
+* Only the ``variants/MQ``, ``calldata/GT``, ``calldata/GQ`` and
+  ``calldata/AD`` fields are required in the output.
+
+* Data types should be as follows: ``variants/MQ``: i1 (single byte
+  signed integer), ``calldata/GT``: i1; ``calldata/GQ``: i1;
+  ``calldata/AD``: i2
+
+* Compressor should be ``zarr.Blosc(cname='zstd', clevel=5,
+  shuffle=-1)``
+
+* Chunk length should be ``2**18``
+
+* ``alt_number`` parameter should be 3 (fix number of alt alleles)
+
+* Data should be grouped by contig (chromosome). This means calling
+  ``vcf_to_zarr()`` once for each contig in the reference genome,
+  specifying the name of the contig as the value of the ``region`` and
+  ``group`` parameters.
+
+Here is an example of the appropriate function call in Python for a
+single contig (this should be called for all contigs in the genome):
+
+```python
+import sys
+import allel
+import zarr
+
+
+def build_sample_zarr(input_path, output_path, contig):
+    allel.vcf_to_zarr(
+        input=input_path,
+        output=output_path,
+        group=contig,
+        region=contig,
+        compressor=zarr.Blosc(cname='zstd', clevel=5, shuffle=0),
+        overwrite=True,
+        fields=['calldata/GT', 'calldata/GQ', 'calldata/AD', 'variants/MQ'],
+        types={'calldata/GT': 'i1',
+               'calldata/GQ': 'i1',
+               'calldata/AD': 'i2',
+	       'variants/MQ': 'i1'},
+        alt_number=3,
+        chunk_length=2**18,
+        log=sys.stdout,
+    )
+
+```
+
+
+### Step: Zip Zarr
+
+Once the zarr conversion is complete, create a zip archive from the
+contents of the root folder of the zarr directory. E.g., if
+``{sample}`` is the sample ID and ``{sample}.zarr`` is the name of the
+output directory used in calls to ``vcf_to_zarr()``, then do:
+
+```bash
+cd {sample}.zarr
+zip -rmT0 {sample}.zarr.zip .
+```
+
+
+## Implementation notes
+
+* For all vector species we currently genotype at all sites in the
+  genome and all possible SNP alleles, which means that the set of
+  alleles provided to the genotyper comprises all genome sites where
+  the reference allele is not "N" and all possible SNP alternate
+  alleles are given in lexical order.
+
+
+## Appendix: Data structures
+
+For the output of the "VCF to Zarr" conversion step, we expect that
+this will create a Zarr hierarchy with the following structure:
+
+* `/` [root group]
+  * `{chromosome}` [group]
+    * `variants` [group]
+      * `MQ` [array; int8; shape (n_sites,)]
+    * `calldata` [group]
+      * `GT` [array; int8; shape (n_sites, 1, 2)]
```

Comment:
Could we can be more explicit on what each dimension represents?  That tripped me up when I first started to work with single-sample zarrs.  Maybe something like:

* `GT` [array; int8; shape (n_sites, samples=1 , ploidy=2)]

or 

* `GT` [array; int8; shape (n_sites, 1 , 2), dimensions correspond to (sites, samples, ploidy)]

Reply:
Thanks, good suggestion.